library/options.inc.php

```diff
@@ -2903,6 +2903,7 @@ function generate_display_field($frow, $currvalue)
             }
         }
     } elseif ($data_type == 54) {
+        global $pid;
```

Comment:
Not sure I like global pid besides there's also an issue with the ContactService. Should be as below removing the 'id' and making a join instead of left join.
```
public function getContactsForPatient($pid, $includeInactive = false): array
    {

        $sql = "SELECT ca.* FROM contact_address ca JOIN contact c ON ca.contact_id = c.id AND c.foreign_table_name = 'patient_data' AND c.foreign_id = ?";
        if ($includeInactive !== true) {
            $sql .= " WHERE ca.status = 'A'";
        }
        $contactData = QueryUtils::fetchRecords($sql, [$pid]) ?? [];
//
//       // does an O(n) fetch from the db, could be optimized later to use populate_array on the record.
        // for single patient pid access this is just fine as very few patients have more than 4-5 addresses
        // TODO: if we need bulk export on patients we should rewrite this method.
        $resultSet = [];
        foreach ($contactData as $record) {
            $contactAddress = new ContactAddress();
            $contactAddress->populate_array($record);
            $address = $contactAddress->getAddress();
            $arrAddress = $address->toArray();
            // overwrite any duplicate values in address with the contact data (the id property will be contact data)
            $arrAddress = array_merge($arrAddress, $contactAddress->toArray());
            $resultSet[] = $arrAddress;
        }
        return $resultSet;
    }
```

Reply:
How about `$addresses = $contactService->getContactsForPatient($_SESSION['pid']) ?? [];` instead in address_list_display.php L-21

Reply:
fine the way it is. it's up to the calling method to ensure correct arguments and scope. anytime globals is called, $pid is in scope.